### DISCUSSION TITLE
Error for missing required parameter

### DISCUSSION THREAD

**Discussion by zxj19910214** [idaholab/moose#28007] [Q&A Getting Started] [answered]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

Dear moose users,

I try to compile input files from:  https://github.com/hugary1995/eel
There is one error I got when I run the input file: 

missing required parameter 'Materials/chemical_energy/reference_chemical_potential'
        Doc String: "reference chemical potential"

The parameter was defined as: 
```
EntropicChemicalEnergyDensity::validParams()
{
  InputParameters params = ChemicalEnergyDensity::validParams();
  params.addRequiredParam<Real>("ideal_gas_constant", "The ideal gas constant");
  params.addRequiredCoupledVar("temperature", "The temperature");
  params.addRequiredCoupledVar(
      "reference_concentration",
      "The reference concentration at which the entropic chemical energy density is zero");
  params.addRequiredParam<MaterialPropertyName>("reference_chemical_potential",
                                                "reference chemical potential");
  return params;
}
```
so it means that this parameter should be used in the input file as it is required as defined, but in the inputfile it is not mentioned, that's why I got the error, right?
```
  [chemical_energy]
    type = EntropicChemicalEnergyDensity
    chemical_energy_density = psi_c
    concentration = c
    ideal_gas_constant = ${R}
    temperature = T
    reference_concentration = c_ref
  []
```
How can I change it to make the simulation run? just try to run this, no need to think about the physics behind.

Thank you for your help in advance.

Best regards,
Xiao

**Comment by zxj19910214** [comment]
I added mu_ref for required parameter <reference_chemical_potential>
first defined it in auxVariables and then initialize it as below:
 [mu_ref_min]
    type = ConstantIC
    variable = mu_ref
    value = 0
    block = 'anode'
  []
  [mu_ref_mid]
    type = ConstantIC
    variable = mu_ref
    value = 0
    block = 'elyte'
  []
  [mu_ref_max]
    type = ConstantIC
    variable = mu_ref
    value = 0
    block ='cathode'
  []

I got error like:
Material property 'mu_ref', requested by 'chemical_energy' is not defined on block anode
Material property 'mu_ref', requested by 'chemical_energy' is not defined on block elyte
Material property 'mu_ref', requested by 'chemical_energy' is not defined on block cathode

**Reply by GiudGiud** [reply, marked as answer]
Hello

> so it means that this parameter should be used in the input file as it is required as defined, but in the inputfile it is not mentioned, that's why I got the error, right?

yes you need to specify this parameter.

For mu_ref, you are getting an error because the chemical_energy requested a material property, and seemingly you added a variable for mu_ref instead

**Reply by zxj19910214** [reply]
yes, thank you, I added the property in the material block, and the error is gone.
